deltacat/compute/compactor/compaction_session.py

```diff
@@ -213,9 +216,10 @@ def _execute_compaction_round(
     # read the results from any previously completed compaction round that used
     # a compatible primary key index
     round_completion_info = None
-    if read_round_completion:
-        logger.info(f"Reading round completion file for compatible "
-                    f"primary key index root path {compatible_primary_key_index_root_path}")
+    if not rebase_source_partition_locator:
+        logger.info(
+            f"Reading round completion file for compatible "
+            f"primary key index root path: {compatible_primary_key_index_root_path}")
         round_completion_info = rcf.read_round_completion_file(
```

Comment:
How would we handle the case when `rebase` itself requires multiple compaction rounds? I think it will be cleaner to separate out `round_completion_files` from `compaction_completion_files`. If the user passes same params, it will re-run all annotated deltas again in an infinite loop.

Reply:
I'll create a separate issue to address this in a follow-up PR, since the current contract of compaction assumes that each round must be able to compact at least one delta. To work with extremely large deltas we'll need to drive that down to at least file-level granularity (which will drive subsequent changes into the Round Completion File and each round that reads it to determine a starting point). Future improvements would then include driving each round down to record-level granularity to work with files that are too large to complete in a single round.

Reply:
Related Issue: https://github.com/ray-project/deltacat/issues/69